cleanlab/segmentation/filter.py

```diff
@@ -0,0 +1,180 @@
+      
+    Tip: If your labels are one hot encoded you can `np.argmax(labels_one_hot,axis=1)` assuming that `labels_one_hot` is of dimension (N,K,H,W)
+    before entering in the function
+
+    pred_probs : np.ndarray
+      An array of shape ``(N,K,H,W,)`` of model-predicted class probabilities,
+      ``P(label=k|x)``. Each pixel contains an array of K classes, where for 
+      an example `x` the array at each pixel contains the model-predicted probabilities 
+      that `x` belongs to each of the K classes. The columns must be ordered such that these probabilities 
+      correspond to class 0, 1, ..., K-1.
+      
+    downsample : int, optional
+      Factor to shrink labels and pred_probs by. Default ``16``
+      Must be a factor divisible by both the labels and the pred_probs. Note that larger factors result in a linear 
+      decrease in performance
+
+    batch_size : int, optional
+      Size of mini-batches to use for estimating the label issues.
+      To maximize efficiency, try to use the largest `batch_size` your memory allows.
+      
+    n_jobs: int, optional
+      Number of processes for multiprocessing (default value = 1). Only used on Linux.
+      If `n_jobs=None`, will use either the number of: physical cores if psutil is installed, or logical cores otherwise.
+    
+    verbose : bool, optional
+      Whether to suppress print statements or not.
+
+    **kwargs:
+        scores_only: optional
+        Set to True to return a score for each image. Meant for internal call in 
+        ``cleanlab.semantic_segmentation.rank.get_label_quality_scores``
+
+    
+    Returns
+    -------
+    label_issues : np.ndarray
+      Returns a boolean **mask** for the entire dataset
+      where ``True`` represents a pixel label issue and ``False`` represents an example that is
+      accurately labeled with high confidence.
+
+    """
+    scores_only = kwargs.get("scores_only", False)
+    
+    def check_input(labels: np.ndarray, pred_probs: np.ndarray) -> None:
+        if len(labels.shape) != 3:
+            raise ValueError("labels must have a shape of (N, H, W)")
+
+        if len(pred_probs.shape) != 4:
+            raise ValueError("pred_probs must have a shape of (N, K, H, W)")
+
+        num_images, height, width = labels.shape
+        num_images_pred, num_classes, height_pred, width_pred = pred_probs.shape
+
+        if num_images != num_images_pred or height != height_pred or width != width_pred:
+            raise ValueError("labels and pred_probs must have matching dimensions for N, H, and W")
+        
+        #Check downsample
+        if height%downsample!=0 or width%downsample!=0:
+            raise ValueError(f"Height {height} and width {width} not divisible by downsample value of {downsample}")
+
+    
+    def downsample_arrays(labels: np.ndarray, pred_probs: np.ndarray, factor: int = 1) -> tuple[np.ndarray, np.ndarray]:
+        if factor == 1:
+            return labels, pred_probs
```

Comment:
```suggestion
    def downsample_arrays(labels: np.ndarray, pred_probs: np.ndarray, factor: int = 1) -> Tuple[np.ndarray, np.ndarray]:
```

Reply:
Fixed ( but not the line removal of return)